### PR TITLE
Add support to override asset (sub) class and url in admin control panel

### LINKED ISSUE #3207: Allow to edit of asset profile in Admin Control panel
As a follow up to https://github.com/ghostfolio/ghostfolio/issues/3151 it should also be possible to edit the `assetClass`, `assetSubClass` and `url` of an asset profile regardless of the data source.

If data source is `MANUAL`, data can still be stored in the `SymbolProfile` database table. For all other data sources, an entry in `SymbolProfileOverrides` needs to be created or updated.

### PR DESCRIPTION
**Pull request by Fedron**
Closes #3207

Allowing editing of asset (sub)class regardless of data source
- If the data source is manual, change is applied directly to the `SymbolProfile`
- Otherwise, a `SymbolProfileOverride` is created with the edited asset (sub)class

Add `url` to asset profile dialog to allow for editing.
Added a link as secondary text to the title of the dialog so that the user can easily navigate to the url.

Allowing editing the url regardless of data source
- If the data source is manual, change is applied directly to the `SymbolProfile`
- Otherwise, a `SymbolProfileOverride` is created with the edited url

### CODE REVIEW

**Comment by Fedron** [comment]
The URL is shown in the dialog here:
image

I'm unsure whether this is a good place for it to be, nor whether the styling is "correct"? Open to suggestions

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts [hunk 1]
```diff
@@ -163,7 +164,8 @@ export class AssetProfileDialog implements OnDestroy, OnInit {
             this.assetProfile?.scraperConfiguration ?? {}
           ),
           sectors: JSON.stringify(this.assetProfile?.sectors ?? []),
-          symbolMapping: JSON.stringify(this.assetProfile?.symbolMapping ?? {})
+          symbolMapping: JSON.stringify(this.assetProfile?.symbolMapping ?? {}),
+          url: this.assetProfile?.url
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts:168]: `url: this.assetProfile?.url ?? ''` ❓

File: apps/api/src/app/admin/admin.service.ts [hunk 2]
```diff
@@ -332,12 +333,18 @@ export class AdminService {
     scraperConfiguration,
     sectors,
     symbol,
-    symbolMapping
+    symbolMapping,
+    url
   }: Prisma.SymbolProfileUpdateInput & UniqueAsset) {
+    const symbolProfileOverride = {
+      name: name as string,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:340]: Can you move down to keep the alphabetical order?

File: apps/api/src/app/admin/admin.service.ts [hunk 3]
```diff
@@ -347,16 +354,12 @@ export class AdminService {
         symbol,
         symbolMapping,
         ...(dataSource === 'MANUAL'
-          ? { name }
+          ? { name, assetClass, assetSubClass, url }
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:357]: `{ assetClass, assetSubClass, name, url }`

File: apps/api/src/app/admin/admin.service.ts [hunk 4]
```diff
@@ -332,12 +333,18 @@ export class AdminService {
     scraperConfiguration,
     sectors,
     symbol,
-    symbolMapping
+    symbolMapping,
+    url
   }: Prisma.SymbolProfileUpdateInput & UniqueAsset) {
+    const symbolProfileOverride = {
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:339]: const symbolProfileOverride**s** = {

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html [hunk 5]
```diff
@@ -215,6 +218,12 @@ <h1 class="flex-grow-1 m-0" mat-dialog-title>
         <input formControlName="name" matInput type="text" />
       </mat-form-field>
     </div>
+    <div class="mt-3">
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html:221]: As `url` it is not that important, can you move it down just before the `comment` field?

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html [hunk 6]
```diff
@@ -7,6 +7,9 @@
   <div class="d-flex mb-3">
     <h1 class="flex-grow-1 m-0" mat-dialog-title>
       {{ assetProfile?.name ?? data.symbol }}
+      <small *ngIf="assetProfile?.url" class="text-body-secondary"
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html:10]: I would remove this 😃

File: apps/api/src/app/admin/admin.service.ts [hunk 7]
```diff
@@ -332,12 +333,18 @@ export class AdminService {
...
+      name: name as string,
+      assetClass: assetClass as $Enums.AssetClass,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:341]: You can use the alias to simplify: `assetClass: assetClass as AssetClass,`

File: apps/api/src/app/admin/admin.service.ts [hunk 8]
```diff
@@ -332,12 +333,18 @@ export class AdminService {
...
+      assetSubClass: assetSubClass as $Enums.AssetSubClass,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:342]: You can use the alias to simplify: `assetSubClass: assetSubClass as AssetSubClass,`

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for the contribution, this is becoming more and more powerful 🤩 I have added a few comments.

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts (hunk 1 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts]: Added

File: apps/api/src/app/admin/admin.service.ts (hunk 2 above)
Comment (Fedron) [reply apps/api/src/app/admin/admin.service.ts:342]: Done

File: apps/api/src/app/admin/admin.service.ts (hunk 3 above)
Comment (Fedron) [reply apps/api/src/app/admin/admin.service.ts]: Done (as a side note is this something prettier can be configured to enforce?)

File: apps/api/src/app/admin/admin.service.ts (hunk 4 above)
Comment (Fedron) [reply apps/api/src/app/admin/admin.service.ts]: Renamed

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html (hunk 5 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html]: Done

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html (hunk 6 above)
Comment (Fedron) [reply apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.html]: Done

File: apps/api/src/app/admin/admin.service.ts (hunk 7 above)
Comment (Fedron) [reply apps/api/src/app/admin/admin.service.ts]: Didn't know that! Changed

File: apps/api/src/app/admin/admin.service.ts (hunk 8 above)
Comment (Fedron) [reply apps/api/src/app/admin/admin.service.ts]: Changed

**Pull request merged**